docs/source/specs/openapi.json

```diff
@@ -705,31 +720,41 @@
             },
             "delete": {
                 "tags": [
-                    "Rate"
+                    "Provider"
                 ],
-                "summary": "Delete a rate",
-                "operationId": "deleteRate",
+                "summary": "Delete a provider",
+                "operationId": "deleteProvider",
                 "parameters": [{
-                    "name": "rate_id",
+                    "name": "uuid",
                     "in": "path",
-                    "description": "ID of rate to get",
+                    "description": "ID of provider to delete",
                     "required": true,
                     "schema": {
                         "type": "string",
-                        "format": "integer"
+                        "format": "uuid"
                     }
                 }],
                 "responses": {
                     "204": {
-                        "description": "Rate deleted"
+                        "description": "Provider deleted"
                     },
                     "401": {
                         "description": "Unauthorized"
                     },
+                    "403": {
+                        "description": "Insufficent permissions to delete provider",
+                        "content": {
+                            "*/*": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/Error"
+                                }
+                            }
+                        }
+                    },
                     "404": {
                         "description": "Not Found",
                         "content": {
-                            "application/json": {
+                            "*/*": {
```

Comment:
Do we ever (under normal circumstances) return a response without json?

Reply:
I don't _think_ so. Koku's exception handler encodes errors as dicts: https://github.com/project-koku/koku/blob/master/koku/api/common/exception_handler.py#L67

And by default uses the json renderer: https://github.com/project-koku/koku/blob/master/koku/koku/settings.py#L260

The content type for errors in the openapi.json is kinda inconsistent. It's mostly `application/json` but there's a couple of `*/*`. I'm not sure if that's an oversight, or if I'm missing something...

Reply:
I think it would be safe to go with `application/json`

Reply:
It's `application/json`
```
> GET /api/cost-management/v1/costmodels/34b2b968-6181-4104-b183-6d47d8243e95/ HTTP/1.1
> Host: 0.0.0.0:8000
> User-Agent: curl/7.61.1
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Date: Tue, 20 Aug 2019 18:33:49 GMT
< Server: WSGIServer/0.2 CPython/3.6.3
< Content-Type: application/json
< Vary: Accept, Origin
< Allow: GET, PUT, PATCH, DELETE, HEAD, OPTIONS
< X-Frame-Options: SAMEORIGIN
< Content-Length: 67
< 
{ [67 bytes data]
100    67  100    67    0     0   3722      0 --:--:-- --:--:-- --:--:--  3722
* Connection #0 to host 0.0.0.0 left intact
{
  "errors": [
    {
      "detail": "Not found.",
      "source": "detail",
      "status": 404
    }
  ]
}
```